packages/handler-graphql/src/responses.ts

```diff
@@ -16,14 +16,29 @@ export class ErrorResponse {
         code: string;
         message: string;
         data?: any;
+        stack?: string;
     };
-    constructor(params: ErrorResponseParams) {
+    constructor(params: ErrorResponseParams | Error | (Error & ErrorResponseParams)) {
         this.data = null;
-        this.error = {
-            code: params.code || defaultParams.code,
-            message: params.message || defaultParams.message,
-            data: params.data || defaultParams.data
-        };
+
+        if (params instanceof Error) {
+            // usually we don't use env variables directly inside code
+            // but otherwise it would make it very tedious to pass this through multiple layers of code
+            const debug = process.env.DEBUG === "true";
+
+            this.error = {
+                code: ("code" in params && params.code) || defaultParams.code,
+                message: params.message || defaultParams.message,
+                data: ("data" in params && params.data) || defaultParams.data,
+                stack: debug ? params.stack : undefined
+            };
+        } else {
+            this.error = {
+                code: params.code || defaultParams.code,
+                message: params.message || defaultParams.message,
+                data: params.data || defaultParams.data
+            };
+        }
     }
```

Comment:
Another thing - we've added the `stack` here in these classes, but not on the GraphQL layer. We want to be able to retrieve this newly added `stack` property via a new `stack` GraphQL field.

Unfortunately we'll need to add this field one by one on all defined `*Error` GraphQL types. You can find these with the following regex search:

![image](https://user-images.githubusercontent.com/5121148/155693768-b624f040-5237-407a-9f8b-e3cd21cc46fc.png)


BTW maybe we could finally introduce a generic `Error` type? Something to think about and potentially tackle in a separate PR.